Improvements to initial implementation

dhontecillas opened the pull request:
Some changes required to have a better interface for the Krakend CE build.

dhontecillas committed:
initial fixes to avoid extra ifs when using otel

deepsource-autofix[bot] committed:
style: format code with Go fmt and Gofumpt

This commit fixes the style issues introduced in 01ffdc6 according to the output
from Go fmt and Gofumpt.

Details: https://github.com/krakend/krakend-otel/pull/4

dhontecillas committed:
fix order of params in GlobalRunServer func

dhontecillas committed:
Merge branch 'v0_2_fixes' of github.com:krakend/krakend-otel into v0_2_fixes

deepsource-autofix[bot] committed:
style: format code with Go fmt and Gofumpt

This commit fixes the style issues introduced in 1984b91 according to the output
from Go fmt and Gofumpt.

Details: https://github.com/krakend/krakend-otel/pull/4

kpacha commented on config/lura.go:
```diff
@@ -96,3 +97,31 @@ func FromLura(srvCfg config.ServiceConfig) (*Config, error) {
+	var cfg *Config
+	var err error
+
+	return func(srvCfg luraconfig.ServiceConfig) (*Config, error) {
```
It's easier to use `sync.Once`, but if we are going to use just a single input, this can be done at the constructor (as in krakend-opencensus and krakend-metrics)

dhontecillas committed:
simplify interfaces by setting a global configuration interface

deepsource-autofix[bot] committed:
style: format code with Go fmt and Gofumpt

This commit fixes the style issues introduced in a583dac according to the output
from Go fmt and Gofumpt.

Details: https://github.com/krakend/krakend-otel/pull/4

dhontecillas committed:
i forgot to commit a file :/

dhontecillas committed:
clean up and improvements of the pipe backend and proxy layers

deepsource-autofix[bot] committed:
style: format code with Go fmt and Gofumpt

This commit fixes the style issues introduced in b7d45b6 according to the output
from Go fmt and Gofumpt.

Details: https://github.com/krakend/krakend-otel/pull/4

dhontecillas committed:
Merge branch 'main' into v0_2_fixes

dhontecillas committed:
fix bug in setting the global config when there is no error in registering instances

dhontecillas committed:
Merge branch 'v0_2_fixes' of github.com:krakend/krakend-otel into v0_2_fixes

dhontecillas committed:
fix deepsource found issues

dhontecillas committed:
better deepsource fixes :)

dhontecillas committed:
rename "http.client.request.read" to "http.client.response.read"
